### PR TITLE
optimize enums

### PR DESCRIPTION
**Pull request by mmoskal**
Not tested at all, but that was my idea

### CODE REVIEW

**Comment by hudson-ai** [comment]
Overall looks pretty good! I think there's a few edge cases we need to watch out for though, particularly when intersecting with other constraints. E.g. intersecting a constant string with a format or pattern may yield an unsatisfiable (sub)schema

File: parser/src/json/schema.rs [hunk 1]
```diff
@@ -128,6 +131,40 @@ impl Schema {
...
+            Schema::Boolean => return Some(RegexNode::Regex("true|false".to_string())),
+            Schema::Number {
+                minimum: Some(x),
+                maximum: Some(y),
+                ..
+            } if x == y => return Some(RegexNode::Literal(x.to_string())),
+            Schema::String {
+                const_string: Some(s),
+                ..
+            } => s,
```
Comment (hudson-ai) [inline parser/src/json/schema.rs:146]: As long as you ensure that there are no other constraints (e.g. due to intersection with siblings) that might invalidate this const_str. Could maybe handle that in the intersection logic, but it should be sufficient to change this to

```rust
            Schema::String {
                min_length: 0,
                max_length: None,
                const_string: Some(s),
                pattern: None,
                format: None,
            } => s,
```

Comment (hudson-ai) [reply parser/src/json/schema.rs:146]: Passes guidance tests (which include some weird cases where other constraints are combined with enums) with this change

Comment (hudson-ai) [reply parser/src/json/schema.rs:146]: but of course this is totally mitigating the entire point of the const, as we're adding a regex when we add the const string... needs modification

Comment (mmoskal) [reply parser/src/json/schema.rs:146]: my idea was that `const_string` means it's just this `const_string`, but it's just a side information, it's also stored as `pattern`; this way the existing intersection logic still works

Comment (mmoskal) [reply parser/src/json/schema.rs:146]: but actually, we can just get rid of const_string completely, and *always* const_compile the ::String!

Comment (mmoskal) [reply parser/src/json/schema.rs:146]: it should be probably regex_compile not const_compile

**Comment by hudson-ai** [comment]
Superceded by #100

**Pull request closed without merging**